vmaas/reposcan/database/csaf_store.py

```diff
@@ -46,7 +48,129 @@ def _save_csaf_files(self, csaf_files: CsafFiles) -> list[int]:
         self.conn.commit()
         return db_ids
 
+    def _get_product_rows(self, csaf_cves):
+        to_insert = []
+        to_update = None
+        for _, csaf_products in csaf_cves.items():
+            for csaf_product in csaf_products:
+                product_values = csaf_product.to_tuple()
+                if not self.csaf_product_map.get(product_values):
+                    to_insert.append(product_values)
+                    # Insert just a key to the map to avoid duplicates.
+                    # Product ID will be updated on refresh cache step after row is inserted into DB.
+                    self.csaf_product_map[product_values] = -1
+
+        return to_insert, to_update
+
+    def _populate_products(self, csaf_cves):
+        self.csaf_product_map = self._prepare_table_map(cols=["cpe", "package", "module"],
+                                                        to_cols=["id"],
+                                                        table="csaf_products")
+
+        self._populate_data(csaf_cves, "csaf_products", ["cpe", "package", "module"],
+                            self._get_product_rows, self._product_refresh_cache,
+                            "(%s::int, %s, %s, %s)")
+
+    def _product_refresh_cache(self, cur, delete=False):
+        for product_id, cpe, package, module in cur.fetchall():
+            if delete:
+                del self.csaf_product_map[(cpe, package, module)]
+            else:
+                self.csaf_product_map[(cpe, package, module)] = product_id
+
+    def _get_cve_rows(self, csaf_cves):
+        # TODO: Known_affected are only parsed as of now. Update here accordingly.
+        product_status_id = 4
```

Comment:
make this configurable with env variable, avoid magic constants in code if possible

Reply:
It should have been done in parser, not in CsafStore class. CsafStore class should just access the data and be done with it for good. When we will start using another product status ids, we will need to refactor CsafStore and the parser as well. Environment variable is not solving that technical debt. I can take care of that, but in separate PR and ticket.

Reply:
then do it in parser